src/engine/cuecontrol.cpp

```diff
@@ -1018,6 +1021,14 @@ HotcueControl::HotcueControl(QString group, int i)
     m_hotcueEnabled = new ControlObject(keyForControl(i, "enabled"));
     m_hotcueEnabled->setReadOnly();
 
+    // NOTE(Swiftb0y): since ControlObjects only deal with double values,
+    // the Color is stored as a QRgb (which is just a unsigned int
+    // representation of the color (AARRGGBB)
+    m_hotcueColor = new ControlObject(keyForControl(i, "color"));
```

Comment:
The current implementation does allow full color customization...
I have settled with Radu's approach to give the border of the marker enough contrast to the marker color. The Database is currently storing the color in the same representation as well.
Regarding the current draft: I'd like to have _atleast_ equal if not more features/possibilites than competitive software (Recordbox, Serato, Tractor, Djay) so If they allow full color customization (I don't know if they do), I want to allow that as well.
If we do settle for the color pallete, how would you store that in the current DB layout, or should we update that to have somesort of color ID instead of an ARGB representation of the color?

Reply:
The idea to use the ARGB representation in DB and ControlObjects works good and is future prove. The only issue is that we IMHO cannot used these "exact" colors on every device, because it might not be able to reproduce it. The other issue is that not all colors are suitable.
So i think it is a good idea to allow  a limited set of known colors.  And allow to alter them depending on the used device/place.

I have searched the web and I think the other tools do the same. 
Serato has 18 colors for cue points and 21 colors for tracks. VDj has 15 colors for cue points. Traktor has 7 colors for tracks and use them to color the whole library line with a dark shade of each color.